### PR TITLE
Add mypy to precommit and github action

### PR DESCRIPTION
**Pull request by ameyakhare**
Resolves AIR-3898

Adds mypy checking to precommit config and as a github action.

### CODE REVIEW

File: airplane/client.py [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
+
+import backoff
+import deprecation
 import requests
 from requests.models import HTTPError
-import uuid
 
-from . import __version__
+from . import __version__  # pylint: disable=cyclic-import
 from .exceptions import InvalidEnvironmentException, RunPendingException
```
Comment (ericliuche) [inline airplane/client.py:12]: relative imports are discouraged, we should use absolute imports

File: airplane/__init__.py [hunk 2]
```diff
@@ -3,55 +3,67 @@
...
+)
+def write_named_output(name: str, value: Any):
     """Writes the value to the task's output, tagged by the key."""
     return _proxy("write_named_output", name, value)
 
 
-def run(task_id, parameters, env={}, constraints={}):
+def run(
+    task_id,
+    parameters,
```
Comment (ericliuche) [inline airplane/__init__.py:55]: do these need to be type annotated?

File: airplane/client.py [hunk 3]
```diff
@@ -36,60 +40,71 @@ def append_output(self, value, *path):
...
         val = json.dumps(value, separators=(",", ":"))
-        self.__chunk_print(f"airplane_output:\"{name}\" {val}")
+        self.__chunk_print(f'airplane_output:"{name}" {val}')
 
-    def __chunk_print(self, output):
-        CHUNK_SIZE = 8192
-        if len(output) <= CHUNK_SIZE:
+    @classmethod
+    def __chunk_print(cls, output: str) -> None:
+        chunk_size = 8192
```
Comment (ericliuche) [inline airplane/client.py:80]: this should be a constant on the class `Airplane.OUTPUT_CHUNK_SIZE`

File: airplane/client.py [hunk 4]
```diff
@@ -1,24 +1,28 @@
...
 class Airplane:
     """Client SDK for Airplane tasks."""
 
-    def __init__(self, api_host, api_token):
+    _api_host: Optional[str]
+    _api_token: Optional[str]
+
+    def __init__(self, api_host: Optional[str], api_token: Optional[str]) -> None:
         self._api_host = api_host
         self._api_token = api_token
```
[the remark this answers is not in the document]
Comment (ameyakhare) [reply airplane/client.py:23]: eh idk if this should be a dataclass, its primary function isnt to represent and store data

File: airplane/__init__.py [hunk 5]
```diff
@@ -3,55 +3,67 @@
...
+
+def set_output(value: Any, *path: Union[str, int]):
     """Sets the task output. Optionally takes a JSON path which can be used
     to set a subpath
     """
     return _proxy("set_output", value, *path)
 
-def append_output(value, *path):
+
+def append_output(value: Any, *path: Union[str, int]):
```
[the remark this answers is not in the document]
Comment (ameyakhare) [reply airplane/__init__.py]: added stricter type checks in mypy section of setup.cfg

File: airplane/client.py [hunk 6]
```diff
@@ -124,7 +136,10 @@ def __backoff():
         ),
         max_tries=1000,
     )
-    def __wait(self, run_id):
+    def __wait(self, run_id: str) -> Dict[str, Any]:
+        if self._api_host is None or self._api_token is None:
+            raise InvalidEnvironmentException()
```
[the remark this answers is not in the document]
Comment (ameyakhare) [reply airplane/client.py:141]: they dont actually need to be valid/not-None if you arent using the run method, so I think its still valid. I agree its a little awkward, but we can address changing that in a separate PR

**Pull request merged**